### PR TITLE
Test Suite Housekeeping (Part 3) 

### PR DESCRIPTION
**Pull request by malefice**
**Note: I estimate that only 1/3 of the transition is complete.**

### What does this change

Since `Faker` also now ships its own `pytest` plugin, this PR aims to (fully or partially) transition the test suite from using `unittest` style test to `pytest` style tests as well as to start using `Faker`'s own fixtures in the test suite. This can also serve as a good opportunity to review the quality of individual tests, and improvements will be made whenever possible.

### CODE REVIEW

**Comment by malefice** [comment]
@fcurella here is a first draft of the pytest transition we discussed in #1162. I would like to hear your thoughts and any suggestion you may have before I proceed with the rest.

File: tests/providers/conftest.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        return faker
+    _class_locale_faker.seed_instance(DEFAULT_SEED)
+    return _class_locale_faker
+
+
+@pytest.fixture(scope='class', autouse=True)
+def num_samples(request):
+    try:
+        num = int(request.cls.num_samples)
+    except:
```
Comment (fcurella) [inline tests/providers/conftest.py:39]: I would use `except AttributeError`, and let it raise `ValueError` if we pass something that's not castable to an `int`

**Comment by malefice** [comment]
@fcurella I think travis hanged since the last merge to master. Anyway, looks like I might be rebasing my PR a fair amount prior to getting this merged if I do not split this now, so I will probably wrap this part up soon and continue with a new PR.

File: tests/providers/conftest.py (hunk 1 above)
Comment (malefice) [reply tests/providers/conftest.py]: Fixed as part of this [commit](https://github.com/joke2k/faker/pull/1167/commits/44f9a3a7975b29247e54e0210b585535d9189866)

**Comment by malefice** [comment]
@fcurella I am done for this portion which (I estimate) accounts for 1/3 of the transition. The tests for other providers are a bit tedious/repetitive/lacking, so I am going to think about how to tackle them, and I also do not want to introduce too much `pytest` magic into the test suite. Anyway, for now, can you please review my work?

**Comment by fcurella** [comment]
@malefice It looks great! Impressive work!

**Pull request merged**